src/dotnet-test-xunit/BinaryWriterTestDiscoverySink.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+using Microsoft.Extensions.Testing.Abstractions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Xunit.Runner.DotNet
+{
+    public class BinaryWriterTestDiscoverySink : ITestDiscoverySink
+    {
+        private readonly BinaryWriter _binaryWriter;
+
+        public BinaryWriterTestDiscoverySink(BinaryWriter binaryWriter)
+        {
+            _binaryWriter = binaryWriter;
+        }
+
+        public void SendTestCompleted()
+        {
+            _binaryWriter.Write(JsonConvert.SerializeObject(new Message
+            {
+                MessageType = "TestRunner.TestCompleted"
```

Comment:
I thought that this was just the runner telling dotnet test that it is done running whatever we asked it to run.